common/core/desktop/src/kmx/kmx_processor.cpp

```diff
@@ -75,43 +106,107 @@ kmx_processor::update_option(
   return option(scope, key, value);
 }
 
+bool
+kmx_processor::queue_action(km_kbp_action_item const * action_item
+) {
+   DebugLog("Action type is [%d].\n", action_item->type);
+  switch (action_item->type) {
+  case KM_KBP_IT_END:
+    // error should not queue empty item
+    return false;
+    break;
```

Comment:
Not sure of the intent of the question.
There is nothing to stop a developer calling queue_action() with `item->type == KM_KBP_IT_END`, it would be silly but nothing to stop it so the code is reachable. Currently it is protected by the for loop in km_kbp_state_queue_action_items.

Reply:
Just `break;` immediately after `return false;` was all I was asking :grin: